src/typechecker.erl

```diff
@@ -1968,7 +1968,7 @@ do_type_check_expr_in(Env, Ty, Cons = {cons, LINE, H, T}) ->
             {     VB2, Cs2} = type_check_expr_in(Env, allow_empty_list(Ty), T),
             {union_var_binds(VB1, VB2, Env#env.tenv), constraints:combine(Cs1, Cs2)};
         {type_error, _} ->
-            throw({type_error, cons, LINE, Cons, Ty})
+            throw({type_error, Cons, type(list), Ty})
     end;
```

Comment:
The actual type is nonempty_list I think. Otherwise, checking a cons against the type [] will give a strange error message.

Reply:
Oh, indeed, I'll fix that later today!